.github/actions/setup-go-tip/action.yml

```diff
@@ -37,5 +39,19 @@ runs:
         echo "GOROOT=$GOROOT" >> $GITHUB_ENV
         echo "GOPATH=$GOPATH" >> $GITHUB_ENV
         echo "PATH=$PATH" >> $GITHUB_ENV
+
+    - name: Build Go Tip from source
+      if: steps.download-go-tip.outputs.download_status == 'false'
+      shell: bash
+      run: |
+        go install golang.org/dl/gotip@latest
+        gotip download
+        echo "GOROOT=$(go env GOROOT)" >> $GITHUB_ENV
+        echo "GOTIP_BIN=$(go env GOPATH)/bin/gotip" >> $GITHUB_ENV
```

Comment:
I see this in your run https://github.com/EraKin575/test-project/actions/runs/9504391060/job/26196935810#step:7:66
```
Success. You may now run 'gotip'!
Run echo "Active Go version:"
Active Go version:
go version go1.22.4 linux/amd[64](https://github.com/EraKin575/test-project/actions/runs/9504391060/job/26196935810#step:7:67)
```
This isn't gotip. And I was expecting that, because you're not setting up env vars the same way as we do in L38-41. For example, you do `GOROOT=$(go env GOROOT)` - that's just default Go version, nothing to do with gotip.

Reply:
![image](https://github.com/jaegertracing/jaeger/assets/100225222/86494a81-8b05-489e-a1fa-82cbf73ba5be)
Someting like this?

Reply:
Note - we could still make it work, I suppose, but then we need to change how the tests are run, e.g. `make test GO=gotip`

Reply:
> Something like this?

I don't like "designing by screenshots" - you should test it out. This is why we have the final step to print go version